core/src/broadcast_stage.rs

```diff
@@ -388,37 +386,26 @@ fn update_peer_stats(
     }
 }
 
-pub fn get_broadcast_peers(
-    cluster_info: &ClusterInfo,
-    stakes: Option<&HashMap<Pubkey, u64>>,
-) -> (Vec<ContactInfo>, Vec<(u64, usize)>) {
-    let mut peers = cluster_info.tvu_peers();
-    let peers_and_stakes = cluster_info::stake_weight_peers(&mut peers, stakes);
-    (peers, peers_and_stakes)
-}
-
 /// broadcast messages from the leader to layer 1 nodes
 /// # Remarks
 pub fn broadcast_shreds(
     s: &UdpSocket,
     shreds: &[Shred],
-    peers_and_stakes: &[(u64, usize)],
-    peers: &[ContactInfo],
+    cluster_nodes: &ClusterNodes<BroadcastStage>,
     last_datapoint_submit: &Arc<AtomicU64>,
     transmit_stats: &mut TransmitShredsStats,
 ) -> Result<()> {
-    let broadcast_len = peers_and_stakes.len();
+    let broadcast_len = cluster_nodes.num_peers();
     if broadcast_len == 0 {
         update_peer_stats(1, 1, last_datapoint_submit);
         return Ok(());
     }
     let mut shred_select = Measure::start("shred_select");
     let packets: Vec<_> = shreds
         .iter()
-        .map(|shred| {
-            let broadcast_index = weighted_best(peers_and_stakes, shred.seed());
-
-            (&shred.payload, &peers[broadcast_index].tvu)
+        .filter_map(|shred| {
+            let node = cluster_nodes.get_broadcast_peer(shred.seed())?;
+            Some((&shred.payload, &node.tvu))
```

Comment:
This should always be `Some` because of the` broadcast_len() == 0` check above right?

Reply:
yeah, but that is an internal implementation detail in `get_broadcast_peer` and here we do not want to rely on that not changing.

Reply:
yup, just confirming my understanding, thanks!